elixir/apps/api/test/api/controllers/gateway_controller_test.exs

```diff
@@ -0,0 +1,200 @@
+    end
+
+    test "returns a single gateway", %{
+      conn: conn,
+      account: account,
+      actor: actor,
+      gateway_group: gateway_group
+    } do
+      gateway = Fixtures.Gateways.create_gateway(%{account: account, group: gateway_group})
+
+      conn =
+        conn
+        |> authorize_conn(actor)
+        |> put_req_header("content-type", "application/json")
+        |> get("/v1/gateway_groups/#{gateway_group.id}/gateways/#{gateway.id}")
+
+      assert json_response(conn, 200) == %{
+               "data" => %{
+                 "id" => gateway.id,
+                 "name" => gateway.name,
+                 "ipv4" => Domain.Types.IP.to_string(gateway.ipv4),
+                 "ipv6" => Domain.Types.IP.to_string(gateway.ipv6)
+               }
+             }
+    end
+  end
+
+  describe "create gateway token" do
+    test "returns error when not authorized", %{conn: conn, gateway_group: gateway_group} do
+      conn = post(conn, "/v1/gateway_groups/#{gateway_group.id}/gateways")
+      assert json_response(conn, 401) == %{"errors" => %{"detail" => "Unauthorized"}}
+    end
+
+    test "creates a gateway token", %{
+      conn: conn,
+      actor: actor,
+      gateway_group: gateway_group
+    } do
+      conn =
+        conn
+        |> authorize_conn(actor)
+        |> put_req_header("content-type", "application/json")
+        |> post("/v1/gateway_groups/#{gateway_group.id}/gateways")
+
+      assert %{"data" => %{"gateway_token" => token}} = json_response(conn, 201)
+
+      assert is_binary(token)
+    end
+  end
+
+  describe "delete" do
+    test "returns error when not authorized", %{
+      conn: conn,
+      account: account,
+      gateway_group: gateway_group
+    } do
+      gateway = Fixtures.Gateways.create_gateway(%{account: account, group: gateway_group})
+      conn = delete(conn, "/v1/gateway_groups/#{gateway_group.id}/gateways/#{gateway.id}")
+      assert json_response(conn, 401) == %{"errors" => %{"detail" => "Unauthorized"}}
+    end
+
+    test "deletes a gateway", %{
+      conn: conn,
+      account: account,
+      actor: actor,
+      gateway_group: gateway_group
+    } do
+      gateway = Fixtures.Gateways.create_gateway(%{account: account, group: gateway_group})
+
+      conn =
+        conn
+        |> authorize_conn(actor)
+        |> put_req_header("content-type", "application/json")
+        |> delete("/v1/gateway_groups/#{gateway_group.id}/gateways/#{gateway.id}")
+
+      assert json_response(conn, 200) == %{
+               "data" => %{
+                 "id" => gateway.id,
+                 "name" => gateway.name,
+                 "ipv4" => Domain.Types.IP.to_string(gateway.ipv4),
+                 "ipv6" => Domain.Types.IP.to_string(gateway.ipv6)
+               }
+             }
+
+      assert {:error, :not_found} ==
+               Gateway.Query.not_deleted()
+               |> Gateway.Query.by_id(gateway.id)
+               |> Repo.fetch(Gateway.Query)
```

Comment:
This is just a style preference but I would do something like this:
```suggestion
      assert gateway = Repo.get(Gateway, gateway.id)
      assert gateway.deleted_at
```
WDYT?

Reply:
Yeah, that's nice.  Will update the delete tests to match that.